src/Run.js

```diff
@@ -16,9 +16,16 @@ function Run(ref, json_q, callback,progressObj) {
       var image = drawarray[pos].image;
       var i = drawarray[pos].i;
       var input = ref.images[image].steps[i - 1].output;
-      ref.images[image].steps[i].draw(ref.copy(input), function onEachStep() {
-        drawStep(drawarray, ++pos);
-      },progressObj);
+
+      generateOutputsArray(input, i, image);
+
+      ref.images[image].steps[i].draw(
```

Comment:
I'd at first be tempted to make a closure here, or to use `bind()` or `apply()` to bind `draw()`'s scope to the local scope and give it access to locally-defined functions. But I believe this would then make it unable to access other local methods within the original Method definition, which is bad - we use them a lot.

So, I think we need to add a local `getStep()` method to the Method object when it's originally created, with a scope of the method instance. Let's see where that ought to go...